sstables/sstables.cc

```diff
@@ -1535,6 +1535,36 @@ future<foreign_sstable_open_info> sstable::get_open_info() & {
     });
 }
 
+future<std::vector<shard_id>>
+sstable::compute_owner_shards(sstable& sst, const io_priority_class& pc) {
+    co_await sst.read_toc();
+    co_await sst.read_scylla_metadata(pc);
+
+    auto has_valid_sharding_metadata = [&sst] {
+        if (!sst.has_component(component_type::Scylla)) {
+            return false;
+        }
+
+        auto& scylla_metadata = sst._components->scylla_metadata;
+        const auto* sm = scylla_metadata
+                ? scylla_metadata->data.get<scylla_metadata_type::Sharding, sharding_metadata>()
+                : nullptr;
+        return sm && sm->token_ranges.elements.size();
+    }();
```

Comment:
Tip for the future: using std::invoke(long_lambda) as opposed to long_lambda() is clearer, since it's immediately visible that you're assigning the result of a calculation rather than naming a lambda. With the other syntax, it's only disambiguated at the end.